presto-main/src/main/java/com/facebook/presto/execution/buffer/ArbitraryOutputBuffer.java

```diff
@@ -76,6 +79,10 @@
     private final AtomicLong totalPagesAdded = new AtomicLong();
     private final AtomicLong totalRowsAdded = new AtomicLong();
 
+    private final ConcurrentMap<Lifespan, AtomicLong> outstandingPageCountPerLifespan = new ConcurrentHashMap<>();
+    private final Set<Lifespan> noMorePagesForLifespan = ConcurrentHashMap.newKeySet();
```

Comment:
Looks like `noMorePagesForLifespan` can only grow as we only `add` and do `contains` checks. Would that be concerning in terms of memory usage? I guess this is true also for `outstandingPageCountPerLifespan`.

Reply:
That's true. It will holding at most all the lifespans on that worker. With 32K lifespans, each worker will have 32K / 700 ~ 500 entries. Is that concerning? -- we can add memory accounting for that if so.